### PR TITLE
fix the bug for metallb operator

### PR DESCRIPTION
**Pull request by alknopfler**
Signed-off-by: Alberto Morgante Medina <alknopfler@users.noreply.github.com>

# Description

Fix the issue with the nmstate operator:
````
account:openshift-nmstate:nmstate-operator" cannot list resource "nodes" in API group "" at the cluster scope
E0713 15:43:47.466609       1 reflector.go:138] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:250: Failed to watch *v1.Node: failed to list *v1.Node: nodes is forbidden: User "system:serviceaccount:openshift-nmstate:nmstate-operator" cannot list resource "nodes" in API group "" at the cluster scope
````

Fixes # reported the bug here: https://bugzilla.redhat.com/show_bug.cgi?id=2106840

## Type of change

Please select the appropriate options:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This change is a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- @achuzhoy tested on his environment working fine after apply the command to fix the issue doing this workarround
 [ ] Test B

**Test Configuration**:

- Versions:
- Hardware:

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not be merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: deploy-metallb/deploy.sh [hunk 1]
```diff
@@ -294,6 +294,11 @@ if ! ./verify.sh; then
         verify_remote_resource ${edgecluster} "default" "crd" "metallbs.metallb.io" "."
         echo
 
+        # "Patch bz https://bugzilla.redhat.com/show_bug.cgi?id=2106840"
+        echo ">> Patching the MetalLB operator"
+        oc --kubeconfig=${EDGE_KUBECONFIG} adm policy add-cluster-role-to-user cluster-admin system:serviceaccount:openshift-nmstate:nmstate-operator
+        oc --kubeconfig=${EDGE_KUBECONFIG} adm policy add-cluster-role-to-user cluster-admin nmstate-operator
```
Comment (achuzhoy) [inline deploy-metallb/deploy.sh:300]: This line is redundant.

Comment (alknopfler) [reply deploy-metallb/deploy.sh]: done

**Comment by alknopfler** [comment]
updated with the ssh command inside instead of using from external host

File: deploy-metallb/deploy.sh [hunk 2]
```diff
@@ -279,6 +279,10 @@ if ! ./verify.sh; then
         copy_files "${EDGE_KUBECONFIG}" "${EDGE_NODE_IP}" "./.kube/config"
         echo
 
+        # "Patch bz https://bugzilla.redhat.com/show_bug.cgi?id=2106840"
+        echo ">> Patching the MetalLB operator"
+        ${SSH_COMMAND} -i ${RSA_KEY_FILE} core@${EDGE_NODE_IP}  "oc adm policy add-cluster-role-to-user cluster-admin system:serviceaccount:openshift-nmstate:nmstate-operator"
```
[the remark this answers is not in the document]
Comment (alknopfler) [reply deploy-metallb/deploy.sh:284]: the only thing we need is kubeconfig, the patch could be applied even without nmstate operator. it's what I'm doing manully.

[the remark this answers is not in the document]
Comment (flaper87) [reply deploy-metallb/deploy.sh:284]: ok, looks like the message we were seeing is "just" a warning. Let's move forward with this for now.

**Review by flaper87 — APPROVED** [review]

**Pull request merged**